libraries/Microsoft.Bot.Connector/Authentication/Retry.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Bot.Connector.Authentication
+{
+    public static class Retry
```

Comment:
First, I would suggest not rolling your own retry and instead consider bringing in a full fledged library such as Polly that is supported by the .NET Foundation.

Second, if you decide you _are_ going to keep this, I would suggest marking it as internal as I don't think you intend this functionality to be leveraged by consumers of your package, right? 🤔

Reply:
Agreed on both points. I might just keep it to unblock customers but this won't survive past 4.3 in favor of Polly.

Reply:
Deferred to 4.3, will definitely look into this!